esmvalcore/preprocessor/_regrid.py

```diff
@@ -502,8 +502,71 @@ def regrid(cube, target_grid, scheme, lat_offset=True, lon_offset=True):
     See Also
     --------
     extract_levels : Perform vertical regridding.
+
+    Notes
+    -----
+    This preprocessor allows for the use of arbitrary :doc:`Iris <iris:index>`
+    regridding schemes, that is anything that can be passed as a scheme to
+    :meth:`iris.cube.Cube.regrid` is possible. This enables the use of further
+    parameters for existing schemes, as well as the use of more advanced
+    schemes for example for unstructured meshes.
+    To use this functionality, a dictionary must be passed for the scheme with
+    a mandatory entry of ``reference`` in the form specified for the object
+    reference of the `entry point data model <https://packaging.python.org/en/
+    latest/specifications/entry-points/#data-model>`_,
+    i.e. ``importable.module:object.attr``. This is used as a factory for the
+    scheme. Any further entries in the dictionary are passed as keyword
+    arguments to the factory.
+
+    For example, to use the familiar :class:`iris.analysis.Linear` regridding
+    scheme with a custom extrapolation mode, use
+
+    .. code-block:: yaml
+
+        my_preprocessor:
+          regrid:
+            target: 1x1
+            scheme:
+              reference: iris.analysis:Linear
+              extrapolation_mode: nanmask
+
+    To use the area weighted regridder available in
+    :class:`esmf_regrid.ESMFAreaWeighted`, make sure that ``iris-esmf-regrid``
+    is installed and use
+
+    .. code-block:: yaml
+
+        my_preprocessor:
+          regrid:
+            target: 1x1
+            scheme:
+              reference: esmf_regrid:ESMFAreaWeighted
+
+    .. note::
+
+        Note that :doc:`iris-esmf-regrid:index` is still experimental.
```

Comment:
maybe add a TODO so we remove this when it's not experimental anymore?

Reply:
Yeah, I was thinking the same thing. In fact, I had even typed it out and then removed it again. Do we have a standard way to note this kind of TODO? I think up to now we only have used issues for that purpose, no?

Reply:
don't think we have any (only for deprecated stuff), maybe just plug one simple one and in 4 years we'll remove it if we don't get nuked in the meantime :grin: